hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/Bound.java

```
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.restjava.service;
+
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.restjava.common.EntityIdRangeParameter;
+import com.hedera.mirror.restjava.common.RangeOperator;
+import java.util.EnumMap;
+import java.util.List;
+import lombok.Getter;
+
+@Getter
+public class Bound {
+
+    private EntityIdRangeParameter lower;
+    private EntityIdRangeParameter upper;
+    private final EnumMap<RangeOperator, Integer> cardinality = new EnumMap<>(RangeOperator.class);
+
+    public Bound(List<EntityIdRangeParameter> params, boolean primarySortField) {
+
+        if (params != null) {
```

switch to fail fast to reduce nesting